derive/src/items.rs

```diff
@@ -107,55 +135,60 @@ fn into_signature(ident: syn::Ident, method_sig: syn::MethodSig, is_constant: bo
 }
 
 fn has_attribute(attrs: &[syn::Attribute], name: &str) -> bool {
-	attrs.iter().any(|a| match a.value {
-		syn::MetaItem::Word(ref ident) => ident.as_ref() == name,
-		_ => false
+	attrs.iter().any(|attr| {
+		if let Some(first_seg) = attr.path.segments.first() {
+			return first_seg.value().ident == name
+		};
+		false
 	})
 }
 
 impl Item {
+	fn event_from_trait_item(method_sig: syn::MethodSig) -> Self {
+		assert!(method_sig.ident.to_string() != "constructor", "The constructor can't be an event");
+		let (indexed, non_indexed) = utils::iter_signature(&method_sig)
+			.partition(|&(ref pat, _)| quote! { #pat }.to_string().starts_with("indexed_"));
+		let canonical = utils::canonicalize_fn(&method_sig.ident, &method_sig);
+		let event = Event {
+			name: method_sig.ident.clone(),
+			canonical: canonical,
+			indexed: indexed,
+			data: non_indexed,
+			method_sig: method_sig,
+		};
+		Item::Event(event)
+	}
+
+	fn signature_from_trait_item(method_trait_item: syn::TraitItemMethod) -> Self {
+		let constant = has_attribute(&method_trait_item.attrs, "constant");
+		let payable = has_attribute(&method_trait_item.attrs, "payable");
+		assert!(!(constant && payable),
+			format!("Method {} cannot be constant and payable at the same time", method_trait_item.sig.ident.to_string()
+		));
```

Comment:
what if we move  `method_trait_item.sig.ident.to_string()` on the next line? E.g.:

```
	assert!(
		!(constant && payable),
		format!(
			"Method {} cannot be constant and payable at the same time",
			method_trait_item.sig.ident.to_string()
		)
	);
```

Reply:
Sure! :)

Reply:
done